docs/config/01-configuration-file.md

```diff
@@ -67,6 +67,45 @@ require('ts-node').register({
 require('./karma.conf.ts');
 ```
 
+### Asynchronous Configuration
+
+The function exported by the configuration file may return a promise whose
+resolution will tell Karma when all configuration is complete, including the
+final call to `config.set()`.
+
+Just as with a synchronous configuration's return value not being used, an
+asynchronous configuration's resolved value will not be used. With both
+configuration types, `config.set()` MUST BE called to assign the configuration
+object.
+
+When using the command line interface, returning a promise is all that is
+needed.
+
+An example of a config that may need asynchronous support is a config that uses
+an async method to find an available port if the preferred port is already in
+use.
+
+```js
+// karma.conf.js
+const findAvailablePort = require('./findAvailablePort.js');
+
+module.exports = function configKarma(config) {
+  const myPreferredPort = 9876;
+  return findAvailablePort(myPreferredPort)
+    .then(function onFreePortResolved(freePort) {
+      config.set({
+        frameworks: ['jasmine'],
+        port: freePort,
+        // ... other Karma configuration
+      });
+    });
+}
+```
+
+If you are using the `parseConfig` method from the public API, then please see
+the [Public API Documentation][dev/public-api]] for details, usage, and
```

Comment:
Extra bracket :)

Reply:
That feeling when you  have checked something 999 times, but didn't go for 1,000.

I have this feeling now! I'll fix it up soon.

Reply:
@jginsburgn - I've fixed the typo and refined the example based on @nicojs 's [feedback in the original PR](https://github.com/karma-runner/karma/pull/3733#pullrequestreview-879956701)